### PR TITLE
feat: Added automatic detection of asynchronous components

### LINKED ISSUE #31: Container decorator
**Is your feature request related to a problem? Please describe.**
When creating dynamic components there is a need to always trigger the `content:loaded` event. This is ok, but could be done better with usage of decorators.

**Describe the solution you'd like**
`@Container` decorator could be used with `@El` to mark element as container for dynamic components. This will use `MutationObserver` scoped in this are and observe changes to DOM to initialise components within scope.

### PR DESCRIPTION
**Pull request by xolir**
fix #31

### CODE REVIEW

File: src/core/linker.js [hunk 1]
```diff
@@ -33,7 +33,11 @@ class Linker {
    */
   link(container = document) {
     Object.keys(this.registry.getData()).forEach((selector) => {
```
Comment (mateuszluczak) [inline src/core/linker.js]: Use also `getRegisteredSelectors` here

File: src/core/init.js [hunk 2]
```diff
@@ -30,6 +31,22 @@ const bindContentEvents = () => {
   });
 };
 
+const onMutationCallback = (mutation) => {
+  const registeredSelectors = registry.getRegisteredSelectors();
+
+  Array.prototype.slice.call(mutation.addedNodes)
+  .filter((node) => {
```
Comment (adamkosmala) [inline src/core/init.js]: why not `.filter((node) => node.nodeName !== 'SCRIPT' && node.nodeType === 1)`?

File: src/core/init.js [hunk 3]
```diff
@@ -30,6 +31,22 @@ const bindContentEvents = () => {
...
+    return node.nodeName !== 'SCRIPT' && node.nodeType === 1;
+  })
+  .forEach((node) => {
+    if (registeredSelectors.find((el) => {
+      return $(node).is(el);
+    })) {
+      bootstrap([node]);
```
Comment (adamkosmala) [inline src/core/init.js]: indent seems to be off

File: src/core/init.js (hunk 2 above)
Comment (panec) [reply src/core/init.js]: or even `.filter(node => node.nodeName !== 'SCRIPT' && node.nodeType === 1)` ?

Comment (xolir) [reply src/core/init.js]: @adamkosmala @panec ESLint rules prevent both of them

File: src/core/init.js (hunk 3 above)
Comment (xolir) [reply src/core/init.js]: It's ok

Comment (adamkosmala) [reply src/core/init.js]: right, did not notice `{` line ealier

**Pull request merged**